OpenRA.Game/Map/Map.cs

```diff
@@ -804,6 +804,17 @@ public CPos ChooseRandomCell(MersenneTwister rand)
 			return new MPos(x, y).ToCPos(this);
 		}
 
+		public CPos ChooseClosestEdgeCell(CPos pos)
+		{
+			var x = pos.X < (Bounds.Right / 2) ? 0 : Bounds.Right;
+			var y = pos.Y < (Bounds.Bottom / 2) ? 0 : Bounds.Bottom;
```

Comment:
Unfortunately that fix won't be enough.  The conditional is also wrong for rectangular maps, and this won't work at all for isometric maps.